### ISSUE TITLE
A problem about deepmedic network classification layer ?

### ISSUE THREAD

**Issue by mjiansun** [Kamnitsask/deepmedic#45]
In [https://github.com/Kamnitsask/deepmedic/blob/master/documentation/deepMedic.png](https://github.com/Kamnitsask/deepmedic/blob/master/documentation/deepMedic.png),
classification layer is 2x9^3.
Question:
1)Does deepmedic predict voxel point category or block 2x9^3 category? If it is 2x9^3, then what is this label？

**Comment by Kamnitsask** [comment]
Hello,

First of all thank you very much for your interest in our work.

I am not sure I understand your question:
Do you mean, why is it that the figure shows two 9^3 classification features maps (a green and a red one)? Instead of a single 9^3 one?
If this is the case:
- It is true that we could have a single feature map there, with a prediction per voxel, giving the probability for each to be lesion. And then, the probability for being healthy would be: P_healthy = 1 - P_lesion. But, this can only be done BINARY classification.
- The reason we have two is to make it follow the generic case of MULTI-CLASS classification. Eg, for 20 classes, you ll have 20 classification-feature maps.
- The output probability maps are [0,1] pseudo-probabilities, after normalizing the logits of these last feature maps with a softmax.

Hope the above answers your question.
Kostas

**Comment by Kamnitsask** [comment]
Now that I read this again, it seems that the question was more about how come the network makes multiple predictions for a single input, instead of predicting only the class for a single voxel per input. This is how fully convolutional networks work, and for more information I ll have to refer you to our paper, Section 2.1.

Regards,
Kostas
